urnx/parse.go

```diff
@@ -0,0 +1,43 @@
+package urnx
+
+import (
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// Parse parses a string into a URN object
+func Parse(urn string) (*URN, error) {
+	conv := strings.Split(urn, ":")
+
+	if conv[0] != prefix {
+		return nil, ErrInvalidURNPrefix
+	}
+
+	ns, err := validateNamespace(conv[1])
+	if err != nil || !ns {
+		return nil, ErrInvalidURNNamespace
+	}
+
+	rt, err := validateResourceType(conv[2])
+	if err != nil || !rt {
+		return nil, ErrInvalidURNResourceType
+	}
+
+	if len(conv) != urnLength {
+		return nil, ErrInvalidURN
+	}
```

Comment:
This should go immediately after the declaration of `conv` or else the above indexing operations might panic.

Reply:
✅